libsel4/include/sel4/bootinfo_types.h

```diff
@@ -69,11 +70,24 @@ typedef struct seL4_BootInfo {
     seL4_SlotRegion   schedcontrol; /* Caps to sched_control for each node */
 #endif
     seL4_SlotRegion   untyped;         /* untyped-object caps (untyped caps) */
-    seL4_UntypedDesc  untypedList[CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS]; /* information about each untyped */
-    /* the untypedList should be the last entry in this struct, in order
-     * to make this struct easier to represent in other languages */
+    seL4_UntypedDesc  untypedList[];   /* information about each untyped */
+    /* the untypedList[] must be the last entry */
 } seL4_BootInfo;
 
+/* The constant CONFIG_MAX_NUM_BOOTINFO_UNTYPED_CAPS has been removed from
+ * CMake configuration and a flexible array untypedList[] is used. It counts a
+ * 0 when using sizeof(seL4_BootInfo), so we can define the max number of
+ * elements according to the boot info frame size here. For userland the
+ * specification defines that the boot info frame is 4 KiB, which corresponds to
+ * the page size on x86, ARM and RISC-V, and extra boot info according to
+ * extraLen/extraBIPages follows afterwards.
+ */
+#define BI_FRAME_SIZE_BITS  12  /* 2^12 = 4096 */
```

Comment:
Why not keep it defined to PAGE_BITS as before? Or seL4_PageBits? It's only 4 kB because that's the smallest page size for all supported platforms.

Isn't this the place that defines the userland specification? It seems strange to refer to it here like this.

(Type "counts as 0".)

Reply:
Thanks, we can use `seL4_PageBits` here. I tried `PAGE_BITS` but that did no work out.